### PR TITLE
Hashing improvements for nested instance types

### PR DESCRIPTION
**Pull request by PJEstrada**
Hashing was not working properly on nested instance so some duplicate cases validations where being missed.

### CODE REVIEW

**Comment by anthony-sarkis** [comment]
Good catch!

File: shared/annotation.py [hunk 1]
```diff
@@ -1176,6 +1189,7 @@ def update_instance(self,
             'center_y': center_y,
             'angle': angle,
             'width': width,
+            'width': width,
```
Comment (anthony-sarkis) [inline shared/annotation.py:1192]: Duplicate?

File: shared/annotation.py [hunk 2]
```diff
@@ -388,14 +388,27 @@ def __check_all_instances_available_in_new_instance_list(self):
...
                                                'Please try reloading page, clicking the refresh file data button or check your network connection. ' \
                                                'Please contact use if this persists.'
-            self.log['error']['missing_ids'] = ids_not_included
-            return False
+            self.log['warning']['missing_ids'] = ids_not_included
+            self.log['warning']['instance_list_new'] = self.instance_list_new
+            self.log['warning']['instance_list_existing_ids'] = [x.id for x in self.instance_list_existing]
+            # TODO: Temporarly removing this hard block since it's causing lots of user experience issue during annotation process
+            # We record this a an event and revisit it in the future
+            Event.new(
```
Comment (anthony-sarkis) [inline shared/annotation.py:403]: Great!

File: frontend/src/components/annotation/annotation_core.vue [hunk 3]
```diff
@@ -6415,15 +6482,20 @@ export default Vue.extend( {
...
+          rear_face: {...inst.rear_face},
           soft_delete: inst.soft_delete,
-          attribute_groups: inst.attribute_groups,
+          attribute_groups: {...inst.attribute_groups},
           machine_made: inst.machine_made,
           sequence_id: inst.sequence_id,
           pause_object: inst.pause_object
-        })
+        }
+        console.log('beffff', inst_data);
```
Comment (anthony-sarkis) [inline frontend/src/components/annotation/annotation_core.vue:6494]: maybe remove logs

File: shared/annotation.py (hunk 1 above)
Comment (PJEstrada) [reply shared/annotation.py]: Fixed now :) thanks for catching

File: shared/annotation.py (hunk 2 above)
Comment (PJEstrada) [reply shared/annotation.py:403]: Thanks Anthony!

File: frontend/src/components/annotation/annotation_core.vue (hunk 3 above)
Comment (PJEstrada) [reply frontend/src/components/annotation/annotation_core.vue]: Done too thank you :)

**Comment by anthony-sarkis** [comment]
Cool :)

**Pull request merged**